app/lib/katello/repo_discovery.rb

```diff
@@ -42,6 +44,70 @@ def run(resume_point)
 
     private
 
+    def parse_parameters(field_value)
+      seen_rel = false
+      has_next_rel = false
+      has_anchor = false
+      while !field_value.empty?
+        field_value.sub!(/^[ \t]*/, '')
+        break if field_value.delete_prefix!(';').nil?
+        field_value.sub!(/^[ \t]*/, '')
+        i = field_value.index(/[ \t=;,]/) || field_value.length
+        parameter_name = field_value.slice!(0, i).downcase(:ascii)
+        field_value.sub!(/^[ \t]*/, '')
+        if field_value.delete_prefix!('=')
+          field_value.sub!(/^[ \t]*/, '')
+          if field_value.delete_prefix!('"')
+            parameter_value = ''
+            while !field_value.empty?
+              break if field_value.delete_prefix!('"')
+              field_value.delete_prefix!("\\")
+              parameter_value += field_value.slice!(0, 1) || ''
+            end
+          else
+            i = field_value.index(/[;,]/) || field_value.length
+            parameter_value = field_value.slice!(0, i)
+          end
+        else
+          parameter_value = ''
+        end
+        case
+        when parameter_name == 'rel' && !seen_rel
+          seen_rel = true
+          has_next_rel = parameter_value.downcase(:ascii).split(/[ \t]/).include?('next')
+        when parameter_name == 'anchor'
+          has_anchor = true
+        end
+      end
+      {:has_next_rel => has_next_rel, :has_anchor => has_anchor}
+    end
+
+    def get_next_link(link_header)
+      # See Appendix B "Algorithms of Parsing Link Header Fiels" of RFC 8288 "Web Linking" (that RFC
+      # appears to be silent about multiple "next" links, so just use the first one and ignore any
+      # additional ones, in the general spirit of being lenient):
+      return nil if link_header.nil?
+      field_value = link_header.clone
+      while !field_value.empty?
+        field_value.sub!(/^[ \t]*/, '')
+        next if field_value.delete_prefix!(',')
+        break if field_value.delete_prefix!('<').nil?
+        i = field_value.index('>')
+        break if i.nil?
+        target_string = field_value[0, i]
+        field_value.slice!(0, i + 1)
+        params = parse_parameters(field_value)
+        if params[:has_next_rel]
+          # To keep it simple, ignore a link with an (unlikely) anchor parameter; but the RFC
+          # mandates that we "MUST NOT process the link without applying the anchor", so just raise
+          # an exception in that (unlikely) case:
+          raise "anchor not supported" if params[:has_anchor]
+          return target_string
```

Comment:
`target_string` could potentially be found more easily with regex:

```ruby
[31] pry(main)> regex = /\<(.*)>/
=> /\<(.*)>/
[32] pry(main)> str = ";<https://website.org?thing=1&otherthing=2>;"
=> ";<https://website.org?thing=1&otherthing=2>;"
[33] pry(main)> str[regex, 1]
=> "https://website.org?thing=1&otherthing=2"
```

Reply:
addressed with 04eecce now

(Though a variant using `$1` and `$'` didn't appear to work and left those empty; but I'm very ignorant of Ruby.  And the quoted `\<` in your sample regexp looks like it wasn't necessary, or was it?)